src/MiddlewareListener.php

```diff
@@ -101,6 +116,42 @@ public function onDispatch(MvcEvent $event)
     }
 
     /**
+     * Create a middleware pipe from the array spec given.
+     *
+     * @param ContainerInterface $serviceLocator
+     * @param ResponseInterface $responsePrototype
+     * @param array $middlewaresToBePiped
+     * @return MiddlewarePipe
+     * @throws \InvalidArgumentException
+     * @throws \Zend\Mvc\Exception\MiddlewareNotCallableException
+     */
+    private function createPipeFromSpec(
+        ContainerInterface $serviceLocator,
+        ResponseInterface $responsePrototype,
+        array $middlewaresToBePiped
+    ) {
+        $pipe = new MiddlewarePipe();
+        $pipe->setResponsePrototype($responsePrototype);
+        foreach ($middlewaresToBePiped as $middlewareToBePiped) {
+            if (null === $middlewareToBePiped) {
+                throw new \InvalidArgumentException('Middleware name cannot be null');
```

Comment:
If we rename the exception to `InvalidMiddlewareException`, we could also add another named constructor: `::becauseNull()` (obviously, choose something better!).